### PR TITLE
578 adding alchemy auto schema

### PR DESCRIPTION
**Pull request by stevbark**
Closes #replace_this_text_with_the_issue_number

### What changes did you make?
  - updated schema.py to used SQLAlchemyAutoSchema.
  -
### Rationale behind the changes?
  - Using SQLAlchemyAutoSchema will make schemas easier to create and maintain. 
  -
### Testing done for these changes
  - Running automated tests
  - Manually checked server endpoints 
### What did you learn or can share that is new?(optional)
  -
  -

### Screenshots of Proposed Changes Of The Website  (if any, please do not screen shot code changes)

<details>
<summary>Visuals before changes are applied</summary>

image

</details>

<details>
<summary>Visuals after changes are applied</summary>
  
image

</details>

### CODE REVIEW

**Comment by stevbark** [comment]
Issue described here: https://github.com/hackforla/HomeUniteUs/issues/578

**Comment by JpadillaCoding** [comment]
Hey @stevbark I'm taking a look into your PR and was trying to figure out a way to test the schemas. How were you approaching testing?

File: api/openapi_server/models/schema.py [hunk 1]
```diff
@@ -1,12 +1,209 @@
...
+        model = ProgramCase
+        include_relationships = True
+        load_instance = True
+
+
+class ProgramCaseStatusLogSchema(SQLAlchemyAutoSchema):
+    class Meta:
+        model = ProgramCaseStatusLog
+        include_relationships = True
+        load_instance = True
```
Comment (Joshua-Douglas) [inline api/openapi_server/models/schema.py:206]: We don't use any of these schema or models. Can we just keep `HostSchema` and `HousingProgramServiceProviderSchema`? Can we also delete all of the corresponding models in the `database.py` file? Our current database model is actually much simpler than what is written, so leaving these models is just confusing.

**Review by Joshua-Douglas — CHANGES_REQUESTED** [review]
Hey @stevbark, 

Thanks for adding this! I agree with @JpadillaCoding, can you write a couple of test cases that:
- demonstrate using marshmallow_sqlalchemy to serialize a Host and HousingProgramServiceProvider database object to JSON
- demonstrate using marshmallow_sqlalchemy to deserialize JSON into a Host and HousingProgramServiceProvider?

It'll be easier to write these types of tests once my PR is merged, but the current pytest proj has most or all of what you need now.

File: .gitignore [hunk 2]
```diff
@@ -3,3 +3,7 @@ data
 **/.db
 .DS_Store
 .vscode
+docker-compose.yml
+api/openapi_server/configs/development.py
+api/openapi_server/configs/staging.py
+app/vite.config.ts
```
Comment (erikguntner) [inline .gitignore:9]: Why are these 4 items being added to the .gitignore?

Comment (stevbark) [reply .gitignore:9]: I had problems with the port. I modified the config on my end and ignored the files so I didnt push the changes.

**Comment by stevbark** [comment]
> Hey @stevbark,
> 
> Thanks for adding this! I agree with @JpadillaCoding, can you write a couple of test cases that:
> 
> * demonstrate using marshmallow_sqlalchemy to serialize a Host and HousingProgramServiceProvider database object to JSON
> * demonstrate using marshmallow_sqlalchemy to deserialize JSON into a Host and HousingProgramServiceProvider?
> 
> It'll be easier to write these types of tests once my PR is merged, but the current pytest proj has most or all of what you need now.

What file should the test be in? In the controller or the Utils test file?

File: api/openapi_server/models/schema.py (hunk 1 above)
Comment (stevbark) [reply api/openapi_server/models/schema.py:206]: Are we sure we want to remove models in the database file? They will probably be required at some point.

File: api/tests/test_schema.py
Comment (stevbark) [inline api/tests/test_schema.py:1]: > Hey @stevbark,
> 
> Thanks for adding this! I agree with @JpadillaCoding, can you write a couple of test cases that:
> 
> * demonstrate using marshmallow_sqlalchemy to serialize a Host and HousingProgramServiceProvider database object to JSON
> * demonstrate using marshmallow_sqlalchemy to deserialize JSON into a Host and HousingProgramServiceProvider?
> 
> It'll be easier to write these types of tests once my PR is merged, but the current pytest proj has most or all of what you need now.

Added in requested tests

File: .gitignore (hunk 2 above)
Comment (erikguntner) [reply .gitignore:9]: Gotcha. Could you revert these changes before we merge? We're still going to want to track changes to these files in the future.

Comment (stevbark) [reply .gitignore]: Update. removed gitignore.

**Comment by stevbark** [comment]
Resolved issues. all tests passed before. Pushed the gitignore update and the tests started failing. not sure why.

**Comment by erikguntner** [comment]
@stevbark yeah that's weird. I re-ran the tests and they passed.

**Review by erikguntner — APPROVED** [review]
This all looks good to me!

**Review by Joshua-Douglas — APPROVED** [review]
Hey @stevbark, 

Approved. The changes look great! It should be a lot easier to (de)serialize our models to and from json now!

The only thing that was missing was the depedency reference in the `pyproject.toml` file. I added in b72eadb632b38936368e551cc4418975572bae2d and recompiled the requirements files using `pip-compile`.

**Pull request merged**